miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -371,6 +377,52 @@ private void validateAliasUniqueness(Sample sample) throws IOException {
     }
   }
 
+  /**
+   * Adds any external names in the sample's attached identity to the identity stored in the database
+   * 
+   * @param sample
+   * @throws IOException
+   */
+
+  private void addExternalNames(DetailedSample sample, SampleIdentity tempIdentity) throws IOException {
+    if (sample.getParent() == null) {
+      throw new IllegalArgumentException(ERR_MISSING_PARENT_ID);
+    }
+    if (tempIdentity == null || tempIdentity.getExternalName() == null) return;
+
+    DetailedSample child = sample;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "child". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)